src/transformStateWithClones.js

```diff
@@ -1,13 +1,26 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
   // write code here
+  const duplicate = { ...state };
+  const array = [];
+
+  for (const action in actions) {
+    if (actions[action]['type'] === 'addProperties') {
```

Comment:
the logic is fine, but code can be much more graceful
use object de-structuring and a switch - case statement with default case
```
const { type, extraData, keysToRemove } = action;

switch (type) {
  case '...':
    ...
    break;
  case '...':
    ...
```